src/main/java/entity/data_structure/DataStore.java

```diff
@@ -0,0 +1,47 @@
+package entity.data_structure;
+
+/**
+ * An abstract data structure.
+ * @param <T> an object for the data structure.
+ */
+public interface DataStore<T> {
+    /**
+     * Adds element to data structure.
+     * @param element value to add to list
+     */
+    void add(T element);
+
+    /**
+     * Removes element from data structure.
+     * @param element element to remove.
+     */
+    void remove(T element);
+
+    /**
+     * Gets element from data structure.
+     * @param element element to get.
+     * @return returns the element.
+     */
+    T get(T element);
```

Comment:
This is great since the interface is not of type Iterable, but we can still get the targeted object. 
Moreover, this leaves the possibility to have some implementations of this interface also implement Itrerable (or not) which leaves a lot of freedom for the future. Nicely done!

However, I believe that it would be a good idea to have a precondition in the java doc that states that the element must exist. Thus, anyone using our API will know that the method will not return the expected output if the element does not exist.